Strip `^O` from the output

teoljungberg opened the pull request:
References #119

teoljungberg commented on autoload/dispatch/tmux.vim:
```diff
@@ -58,7 +58,7 @@ function! dispatch#tmux#make(request) abort
     let filter .= ' -u'
   endif
-  let filter .= " -e \"s/\r$//\" -e \"s/.*\r//\" -e \"s/\e\\[K//g\" -e \"s/.*\e\\[2K\e\\[0G//g\" -e \"s/\e\\[[0-9;]*m//g\" > ".a:request.file
+  let filter .= " -e \"s/\r$//\" -e \"s/.*\r//\" -e \"s/\e\\[K//g\" -e \"s/.*\e\\[2K\e\\[0G//g\" -e \"s/\e\\[[0-9;]*m//g\" -e \"s///g\" > ".a:request.file
```
Is there a better way of accomplishing this in `sed`?

tpope replied:
No, but you can definitely do better than putting a binary control character in a Vim script.

teoljungberg replied:
I was skeptical over that too, could you give an example how this can be solved?

tpope replied:
Like most other languages, VimL strings support octal escapes.

tpope commented:
Why?

teoljungberg commented:
The rust compiler outputs `^O`, which are visible when using the tmux adapter.
This change strips those characters out before reading the output into vim's quickfix window.

Screenshots to demonstrate this are posted in #119

tpope commented:
That's more of a what than a why. Best I can tell it's a no-op to disable a mode that isn't enabled: https://en.wikipedia.org/wiki/Shift_Out_and_Shift_In_characters. So yeah, a good candidate to strip out.

teoljungberg commented:
@tpope I updated the PR to use `nr2char()` to remove the octal character.
I also refactored the variable `filter` to be split up into multiple lines, which might or might not be something we want to do.

Let me know if there's anything else I can do.

tpope commented:
Use an octal escape sequence, and squash.

teoljungberg commented:
Updated and squashed

justinmk commented on autoload/dispatch/tmux.vim:
```diff
@@ -58,7 +58,12 @@ function! dispatch#tmux#make(request) abort
+  let filter .= " -e \"s/\e\\[K//g\" "
+  let filter .= " -e \"s/.*\e\\[2K\e\\[0G//g\""
+  let filter .= " -e \"s/\e\\[[0-9;]*m//g\""
+  let filter .= " -e \"s/\<C-O>//g\""
```
`\017` is the octal

teoljungberg replied:
Good call!
I found that this reads clearer, since it shows more explicitly which key we are stripping out.

What do you think?

tpope replied:
It's "shift in", though, not somebody banging Control+O on a keyboard.

teoljungberg replied:
Updated to use the octal escape now.

> On 29 Jan 2016, at 17:51, Tim Pope <notifications@github.com> wrote:
> 
> In autoload/dispatch/tmux.vim:
> 
> > @@ -58,7 +58,12 @@ function! dispatch#tmux#make(request) abort
> >    elseif uname ==# 'Linux'
> >      let filter .= ' -u'
> >    endif
> > -  let filter .= " -e \"s/\r$//\" -e \"s/.*\r//\" -e \"s/\e\\[K//g\" -e \"s/.*\e\\[2K\e\\[0G//g\" -e \"s/\e\\[[0-9;]*m//g\" > ".a:request.file
> > +  let filter .= " -e \"s/\r$//\" -e \"s/.*\r//\""
> > +  let filter .= " -e \"s/\e\\[K//g\" "
> > +  let filter .= " -e \"s/.*\e\\[2K\e\\[0G//g\""
> > +  let filter .= " -e \"s/\e\\[[0-9;]*m//g\""
> > +  let filter .= " -e \"s/\<C-O>//g\""
> 
> It's "shift in", though, not somebody banging Control+O on a keyboard.
> 
> —
> Reply to this email directly or view it on GitHub.
>
